plugins/catalog-backend/src/permissions/rules/createPropertyRule.ts

```diff
@@ -15,16 +15,26 @@
  */
 
 import { get } from 'lodash';
-import { Entity } from '@backstage/catalog-model';
 import { RESOURCE_TYPE_CATALOG_ENTITY } from '@backstage/plugin-catalog-common';
 import { createCatalogPermissionRule } from './util';
+import { z } from 'zod';
 
 export const createPropertyRule = (propertyType: 'metadata' | 'spec') =>
-  createCatalogPermissionRule({
+  createCatalogPermissionRule<{
+    key: string;
```

Comment:
Why is it necessary to explicitly pass TParams here?

Reply:
Ah it was to correctly reflect the optional parameters, however using `z.input` for typing the toQuery and apply functions fixed this.